### PR TITLE
Animas bug fix: Final basal when rate not yet zero after suspend

### PR DESCRIPTION
**Pull request by gniezen**
Fixes an edge case where at the time of upload the basal rate hasn't changed to zero yet. (It takes up to three minutes for the basal rate to change after a command is issued on the pump). 

This PR calculates the duration of the final basal based on the time of the suspend event, so that it can be displayed.

### CODE REVIEW

**Comment by gniezen** [comment]
@jebeck will you be able to review?

File: lib/animas/animasSimulator.js [hunk 1]
```diff
@@ -153,7 +153,7 @@ exports.make = function(config){
           // categorize the basal we're processing as a suspend basal if its rate
           // is zero and its start falls within the suspend event we have on deck
           var resumed = Date.parse(currStatus.time) + currStatus.duration;
-          if (resumed > Date.parse(event.time)) {
+          if (resumed >= Date.parse(event.time)) {
```
Comment (jebeck) [inline lib/animas/animasSimulator.js]: Why the change to `>=` here? I'm not quite following.

File: lib/drivers/animasDriver.js [hunk 2]
```diff
@@ -2085,6 +2085,7 @@ module.exports = function (config) {
             debug('[Hand-off to simulator] Unhandled type!', datum.type);
         }
       }
+      simulator.finalBasal();
```
Comment (jebeck) [inline lib/drivers/animasDriver.js]: So, thinking out loud here to check my understanding... We didn't have a `finalBasal` method in the simulator before now analogous to the method used for OmniPod & Tandem (looking up settings as of upload and guessing at duration of basal) because you have to suspend the pump in order to upload an Animas, correct? And in the "normal" case (not the case this PR is about), the pump *has* already gone to rate `0` by upload and so that suspend segment would be the final basal, and so we don't have any way to guess about the duration?

File: lib/animas/animasSimulator.js (hunk 1 above)
Comment (gniezen) [inline lib/animas/animasSimulator.js]: Not completely relevant to this PR, but as it was a one-character change I didn't want to create a whole separate PR. Basically, since we only have one-minute resolution, the suspend/resume event and the basal can stop at exactly the same time (but they can also be up to three minutes apart).

File: lib/drivers/animasDriver.js (hunk 2 above)
Comment (gniezen) [inline lib/drivers/animasDriver.js]: Exactly. 💯

File: lib/animas/animasSimulator.js (hunk 1 above)
Comment (jebeck) [inline lib/animas/animasSimulator.js]: Ah, OK. I'm glad it's not completely related because that's what was confusing me most - trying to see the connection with the other changes.

**Comment by jebeck** [comment]
My questions have been answered, so :shipit:

**Comment by gniezen** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;5Q1QHuWU&#x2F;5-ux-0-dev-3-tst-2-uploader-animas-remaining-work-on-animas"> &lt;ux:0 dev:3 tst:2&gt; [uploader] [animas] remaining work on Animas</a>
